### PR TITLE
RYA-174 Removing Mongo Test Factory from non-test code

### PR DESCRIPTION
**Pull request by amihalik**
> What Changed?

Removed Mongo Test Factory code from non-test code
### Tests

no new tests
### Links

[Jira](https://issues.apache.org/jira/browse/RYA-174)
### Checklist
- [ ] Code Review
- [ ] Squash Commits
#### People To Reivew

@dlotts

### CODE REVIEW

**Comment by isper3at** [comment]
I disagree with the notion of matching accumulo naming conventions. I'm fine with ambiguous naming in interfaces and during design, but altering naming to match another datastore just seems.....wrong?  If we want consistency, lets use what flapdoodle calls it.  Then we already have the correct naming if we need to google something.

**Comment by pujav65** [comment]
hey i see where this was coming from, but i think we should try to move to the mongo client being passed in.  I'd like to see this look a lot closer to what was done in Accumulo (more because the Accumulo implementation is in line with what the intended contracts of the RyaDAO/etc. are).   We started moving away from that when we started having a connections factory -- I think this pull request removes that which is good, but I agree with Aaron that we should move to have the connector/client be passed in to everything that needs it rather than passing in the connection information.

**Comment by isper3at** [comment]
what do you mean by 'removes that which is good'?  The connections factory is a placeholder until we can get rid of the dependency on having the hadoop configuration everywhere (i put it in the docs the first time it went in).  Accumulo has had a factory style object for a while now, the AccumuloInstanceDriver is essentially a factory

File: dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/MockMongoFactory.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import de.flapdoodle.embed.mongo.config.MongodConfigBuilder;
+import de.flapdoodle.embed.mongo.config.Net;
+import de.flapdoodle.embed.mongo.config.RuntimeConfigBuilder;
+import de.flapdoodle.embed.mongo.distribution.IFeatureAwareVersion;
+import de.flapdoodle.embed.mongo.distribution.Version;
+
+public class MockMongoFactory {
+    private static Logger logger = LoggerFactory.getLogger(MockMongoFactory.class.getName());
+
+    public static MockMongoFactory newFactory() throws IOException {
```
Comment (amihalik) [inline dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/MockMongoFactory.java]: hrm.  I couldn't find that class.  I won't consolidate the classes in this PR.  Let's do it in another Jira Issue.

File: dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/MockMongoFactory.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        }
+    }
+
+    /**
+     * Creates a new Mongo connection.
+     * 
+     * @throws MongoException
+     * @throws UnknownHostException
+     */
+    public MongoClient newMongo() throws UnknownHostException, MongoException {
```
Comment (amihalik) [inline dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/MockMongoFactory.java]: Done.

File: dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/MongoConnectorFactory.java [hunk 3]
```diff
@@ -56,28 +52,14 @@
...
-     */
-    private static void createMongoClientForTests() throws MongoException {
-        try {
-            MongodForTestsFactory testsFactory = MongodForTestsFactory.with(Version.Main.PRODUCTION);
-            mongoClient = testsFactory.newMongo();
-        } catch (IOException e) {
-            // Rethrow as an unchecked error.  Since we are in a test mode here, just fail fast.
-            throw new MongoException(MSG_INTRO+"creating a factory for a test/mock MongoDB instance.",e);
-        }
+    public static synchronized void closeMongoClient() {
```
Comment (amihalik) [inline dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/MongoConnectorFactory.java]: Done.

File: dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/MockMongoFactory.java (hunk 1 above)
Comment (isper3at) [inline dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/MockMongoFactory.java]: Oh, its not in yet.  Its in my RYA-123 pull request

**Comment by amihalik** [comment]
It seems like everyone is on board with this PR... I'm going to let Jenkins chomp on it for a bit and (unless I hear otherwise) pull this into master after the build.

**Comment by pujav65** [comment]
i thought the mongo connections factory was a bad idea at the time.  so i think removing it is a good thing in general.  The instance driver you're referring to is in the merge tool -- its not used in the main client path.  The accumulo dao has a setConnector() method which really should have been emulated by the Mongo dao.   I didn't realize the pattern until we started having issues with setting up multiple connections, but the passing in the connector/client pattern is nice because its a lot more flexible than what we currently have.  Regardless, I think this pull request is fine -- all of this discussion is on future direction.

**Pull request closed without merging**